src/processing/UserUtil.java

```diff
@@ -0,0 +1,135 @@
+	 * @param username Username
+	 * @param ipAddress Server's ip address
+	 * @param boardId Board ID	
+	 */
+    public static void initiateUserConnection(
+    		String username, 
+    		String ipAddress, 
+    		String boardId
+    	) {
+    	
+    	// stores the information
+    	ClientBoardState.username  = new Username(username);
+    	ClientBoardState.serverIp  = new IpAddress(ipAddress);
+    	ClientBoardState.boardId   = new BoardId(boardId);
+    	
+    	// gets the client information (ip address + free port)
+    	ClientInfo clientInfo = CommunicatorFactory.getClientInfo();
+    	logger.log(ModuleID.PROCESSING, LogLevel.SUCCESS, "Got the client information");
+    	
+    	// stores the user ip and port number
+    	ClientBoardState.userIp  = new IpAddress(clientInfo.getIp());
+    	Port userPort  = new Port(clientInfo.getPort());
+    	ClientBoardState.userPort = userPort;
+    	
+    	// gets a communicator and starts it
+    	ClientBoardState.communicator = CommunicatorFactory.getCommunicator(userPort.port);
+    	ClientBoardState.communicator.start();
+    	// starts the client side processing
+    	ClientBoardState.start();
+    	
+    	logger.log(
+    			ModuleID.PROCESSING, 
+    			LogLevel.SUCCESS, 
+    			"Started communicator and the ClientBoardState"
+    	);
+    	return;
+    }
+    
+    /**
+     * Finds the username of the objects drawn at these list of positions
+     * 
+     * Since, there could be multiple objects at these positions drawn by different
+     * users so username of the user having maximum occurrences at these positions
+     * is selected
+     * 
+     * @param positions list of positions
+     * @return username
+     * @return null if there is no object present at those positions
+    */
+    
+    public static String getUser(ArrayList <Position> positions) {
+    	
+    	// null object
+    	if (positions == null) {
+    		logger.log(ModuleID.PROCESSING, LogLevel.WARNING, "Got a null ArrayList in getUser");
+    		return null;
+    	}
+    	
+    	// gets the id of the object occurring maximum times at these positions
+    	ObjectId objectId = ClientBoardState
+    						.maps
+    						.getMostProbableObjectId(positions);
+    	
+    	String maxOccUser = null;
+    	
+    	// if a object found
+    	if (objectId != null) {
+    		// gets the username for the above object
+    		maxOccUser = ClientBoardState
+    					.maps
+    					.getUserNameOfObject(objectId);
+    	}
+    	
+    	logger.log(
+    			ModuleID.PROCESSING, 
+    			LogLevel.SUCCESS, 
+    			"Got the maximum occuring user"
+    	);
+    	return maxOccUser;
+    }
+    
+    /**
+     * Stops the board session for the user
+     */
+    public static void stopBoardSession() {
+    	/**
+    	 * Informs the server about closing the session of the user
+    	 * 
+    	 * identifier is "StopConnection"
+    	 * message is userId
+    	 * destination is the server's address
+    	 */
+    	ClientBoardState.communicator.send(
```

Comment:
need to call stop from IServerCommunication instead of you sending.

Reply:
Done